### PR TITLE
Add the exact Wald solution to ForceFree system

### PR DESCRIPTION
**Pull request by yoonso0-0**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/PointwiseFunctions/AnalyticSolutions/ForceFree/ExactWald.cpp [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "PointwiseFunctions/InitialDataUtilities/InitialData.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace ForceFree::Solutions {
+
+ExactWald::ExactWald(const double magnetic_field_amplitude,
+                     const Options::Context& /*context*/)
```
Comment (wthrowe) [inline src/PointwiseFunctions/AnalyticSolutions/ForceFree/ExactWald.cpp]: [optional] If you don't use the context you can omit the argument and the option parser will figure out not to pass it.

File: src/PointwiseFunctions/AnalyticSolutions/ForceFree/ExactWald.cpp [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  return std::make_unique<ExactWald>(*this);
+}
+
+void ExactWald::pup(PUP::er& p) {
+  InitialData::pup(p);
+  p | background_spacetime_;
+  p | magnetic_field_amplitude_;
+}
+
+PUP::able::PUP_ID ExactWald::my_PUP_ID = 0;
```
Comment (wthrowe) [inline src/PointwiseFunctions/AnalyticSolutions/ForceFree/ExactWald.cpp:40]: Just NOLINT this to make clang-tidy happy.  We can't do anything about charm internals.

File: src/PointwiseFunctions/AnalyticSolutions/ForceFree/ExactWald.hpp [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+ * $E_iB^i = 0$ is still satisfied, but $B^2 > E^2$ is not.
+ *
+ */
+class ExactWald : public evolution::initial_data::InitialData,
+                  public MarkAsAnalyticSolution {
+ public:
+  struct MagneticFieldAmplitude {
+    using type = double;
+    static constexpr Options::String help = {
+        "Amplitude of initial magnetic field along z axis."};
```
Comment (wthrowe) [inline src/PointwiseFunctions/AnalyticSolutions/ForceFree/ExactWald.hpp]: This is time-independent, so remove "initial".

File: src/PointwiseFunctions/AnalyticSolutions/ForceFree/ExactWald.hpp [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace ForceFree::Solutions {
+
+/*!
+ * \brief An exact electrovacuum solution of Maxwell's equations in the Kerr
+ * spacetime found by Wald \cite Wald1974.
```
Comment (wthrowe) [inline src/PointwiseFunctions/AnalyticSolutions/ForceFree/ExactWald.hpp]: Kerr -> Schwarzschild?  The paper is valid for arbitrary spin, but only the zero-spin case is implemented (and I take it that the spinning case isn't a solution to the force-free equations.)

File: src/PointwiseFunctions/AnalyticSolutions/ForceFree/Factory.hpp [hunk 5]
```diff
@@ -4,12 +4,13 @@
...
+#include "PointwiseFunctions/AnalyticSolutions/ForceFree/ExactWald.hpp"
 #include "PointwiseFunctions/AnalyticSolutions/ForceFree/FastWave.hpp"
 #include "Utilities/TMPL.hpp"
 
 namespace ForceFree::Solutions {
 /*!
  * \brief Typelist of all analytic solutions of GRFFE evolution system
  */
-using all_solutions = tmpl::list<AlfvenWave, FastWave>;
+using all_solutions = tmpl::list<AlfvenWave, FastWave, ExactWald>;
```
Comment (wthrowe) [inline src/PointwiseFunctions/AnalyticSolutions/ForceFree/Factory.hpp]: Alphabetize.

File: tests/Unit/PointwiseFunctions/AnalyticSolutions/ForceFree/Test_ExactWald.cpp [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+  test_move_semantics(ExactWald{1.0}, ExactWald{1.0});
+
+  // test derived
+  register_classes_with_charm<ExactWald>();
+  const std::unique_ptr<InitialData> base_ptr =
+      std::make_unique<ExactWald>(0.5);
+  const std::unique_ptr<InitialData> deserialized_base_ptr =
+      serialize_and_deserialize(base_ptr)->get_clone();
+  CHECK(dynamic_cast<const ExactWald&>(*deserialized_base_ptr.get()) ==
+        dynamic_cast<const ExactWald&>(*base_ptr.get()));
```
Comment (wthrowe) [inline tests/Unit/PointwiseFunctions/AnalyticSolutions/ForceFree/Test_ExactWald.cpp]: Remove `.get()` form these two lines.

**Review by wthrowe — CHANGES_REQUESTED** [review]
The other clang-tidy warning is legitimate.

File: src/PointwiseFunctions/AnalyticSolutions/ForceFree/ExactWald.hpp (hunk 4 above)
Comment (yoonso0-0) [reply src/PointwiseFunctions/AnalyticSolutions/ForceFree/ExactWald.hpp]: Updated the docs

**Comment by wthrowe** [comment]
Looks good.  Squash.

**Comment by yoonso0-0** [comment]
Squashed and rebased on develop. Thanks for review.

**Review by wthrowe — APPROVED** [review]

**Pull request merged**